Cargo.toml

```diff
@@ -51,6 +51,7 @@ hostname-validator = "1.1.1"
 reqwest = { version = "0", default-features = false, features = [
   "rustls-tls-native-roots",
 ] }
+rustls = { version = "0.23", default-features = false }
```

Comment:
Have you tried to relax that even more, like `version = "0"`. This should avoid any complication with the next minor release of rustls